diff --git a/qcwcn/wpa_supplicant_8_lib/driver_cmd_nl80211_common.h b/qcwcn/wpa_supplicant_8_lib/driver_cmd_nl80211_common.h
index 0e532127b173c9845e1c1baa9c2363fc2a0a7171..75d70d52460e6f19b1496ba79f3fa7febd17fd2b 100644
--- a/qcwcn/wpa_supplicant_8_lib/driver_cmd_nl80211_common.h
+++ b/qcwcn/wpa_supplicant_8_lib/driver_cmd_nl80211_common.h
@@ -54,8 +54,9 @@
 #define MAX_NUM_MLO_LINKS      15
 #define NL80211_ATTR_MAX_INTERNAL 256
 
-struct wpa_driver_nl80211_data *drv;
-struct i802_bss *bss;
+struct wpa_driver_nl80211_data;
+struct i802_bss;
+
 struct nl_msg *prepare_nlmsg(struct wpa_driver_nl80211_data *drv,
 		             char *ifname, int cmdid, int subcmd, int flag);
 struct nl_msg *prepare_vendor_nlmsg(struct wpa_driver_nl80211_data *drv,
